explorer/interpreter/interpreter.cpp

```diff
@@ -2210,6 +2210,15 @@ auto Interpreter::StepDeclaration() -> ErrorOr<Success> {
     case DeclarationKind::VariableDeclaration: {
       const auto& var_decl = cast<VariableDeclaration>(decl);
       if (var_decl.has_initializer()) {
+        auto initializer = &var_decl.binding().static_type();
+        if (const auto* dest_class = dyn_cast<NominalClassType>(initializer)) {
+          if (dest_class->declaration().extensibility() ==
+              ClassExtensibility::Abstract) {
+            return ProgramError(var_decl.source_loc())
+                   << "Cannot instantiate abstract class "
+                   << dest_class->declaration().name();
+          }
```

Comment:
Please can you factor this out into a separate function, perhaps `ExpectNonAbstractType` or `ExpectConcreteType`? I expect we'll want to do this from a few different places in the type checker.

Reply:
Done